### DISCUSSION TITLE
Will TreeDataGrid ever be added into the main repo?

### DISCUSSION THREAD

**Discussion by dariostrm** [AvaloniaUI/Avalonia#16617] [General]
I Love using the FluentAvalonia style and controls, but it doesn't have styles for the TreeDataGrid because it isn't in the main avalonia repo, even though it's a very useful control imo...

**Comment by stevemonaco** [comment]
There's nothing technical that prevents FluentAvalonia from distributing its own `TreeDataGrid` theme, AFAIK. `TreeDataGrid` is distributed in a separate package, similar to `DataGrid`. Versioning can lag behind in breaking ways due to the separate repo, but such `ControlTheme` could be distributed as a separate package or simply source.

`TreeDataGrid` has a much lesser focus due to https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid/issues/98

**Comment by maxkatz6** [comment]
Also, TreeDataGrid doesn't depend on Fluent (non FluentAvalonia) theme, it is more flexible.

You can reference [avares://Avalonia.Controls.TreeDataGrid/Themes/Generic.axaml](https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid/blob/master/src/Avalonia.Controls.TreeDataGrid/Themes/Generic.axaml) and it will work with any theme. You can also override default brushes as well.

**Reply by dariostrm** [reply]
Awesome, thank you
